### DISCUSSION TITLE
Video Cut while playing the RTSP Stream

### DISCUSSION THREAD

**Discussion by AVResearch2** [SuRGeoNix/Flyleaf#488] [Q&A]
we are using   Flyleaf library to play a RTSP stream in WPF application ,  we are recoding the stream while its playing using  Player.StartRecording(ref destFile,false); ,    we need to create a video clips based on start time and duration after this part of the video is recorded  while the stream is continue playing and recording 

is there any support / method available in Flyleaf library to initiate video clip cutting , 

would appreciate any solutions or  suggestion regarding this.
Thanks in advance.

**Comment by SuRGeoNix** [comment]
Generally, there is no build-in way to do this. I would look into the [Downloader](https://github.com/SuRGeoNix/Flyleaf/blob/master/FlyleafLib/MediaFramework/MediaContext/Downloader.cs) might be able to use this directly be seeking the demuxer and disposing when the CurTime reaches the desired duration. In any case you could create your own 'Downloader' for your purpose. There is a new library coming up that will allow you to perform more advanced media editing as this one was mainly for playback.
